### DISCUSSION TITLE
Export pressure as one of the attributes?

### DISCUSSION THREAD

**Discussion by junyansim** [InteractiveComputerGraphics/SPlisHSPlasH#142] [Q&A] [answered]
Is it possible to export pressure as one of the attributes? I try adding 
"particleAttributes": "velocity;density;pressure"
in the scene file. However, it seems like it will give me the same attributes even if I change the name of the third attributes to any name. So I am not sure whether the third attribute is really the pressure.

**Comment by janbender** [comment, marked as answer]
Yes, this is possible. You can export any particle attribute that exists. All attribute names can be found in the GUI, tab "Fluid Model" -> "Fluid" -> "Color field" since the attributes can also be used for coloring the particles. Alternatively you could select one particle in the simulation and press the key "i" (information) to get all attribute keys and values. 

However, the pressure attribute does not exist for all pressure solvers, e.g. DFSPH only provides kappa and kappa_v where:

kappa = pressure / density

kappa is used by the constant density solver and kappa_v by the divergence solver.

**Reply by junyansim** [reply]
Thank you very much for taking your time to answer my question! 

I would like to ask a follow up question. Since there are two pressures in DFSPH, is it appropriate to add the two pressures to get the overall pressure and compare with the pressure in other solvers such as IISPH or other Eulerian solvers?

**Reply by janbender** [reply]
I'm not so sure if this is a good idea to simply add the pressures and to compare it with other solvers. DFSPH solves two PPEs to enforce a constant density and a divergence-free velocity field (https://animation.rwth-aachen.de/publication/0559/). IISPH only enforces a constant density. In fact IISPH is equivalent to DFSPH with a disabled divergence-free solve. Most Eulerian solvers only enforce a divergence-free velocity field. Therefore, they often suffer from volume loss, especially when using large step sizes. This problem can be solved by a Eulerian constant density solve (https://animation.rwth-aachen.de/publication/0566/).
